core/src/main/scala/dagr/core/tasksystem/Task.scala

```diff
@@ -23,20 +23,167 @@
  */
 package dagr.core.tasksystem
 
-import com.fulcrumgenomics.commons.CommonsDef.unreachable
+import java.time.{Duration, Instant}
+
+import com.fulcrumgenomics.commons.CommonsDef.{FilePath, unreachable}
+import com.fulcrumgenomics.commons.io.PathUtil
+import com.fulcrumgenomics.commons.util.TimeUtil.formatElapsedTime
+import dagr.api.models.{ResourceSet, TaskStatus, TimePoint}
+import dagr.core.DagrDef.TaskId
+import dagr.core.exec.Executor
 import dagr.core.execsystem.TaskExecutionInfo
+import dagr.core.tasksystem.Task.TaskInfo
 
 import scala.collection.mutable
 import scala.collection.mutable.ListBuffer
+import scala.reflect.ClassTag
 import scala.util.control.Breaks._
 
 /** Utility methods to aid in working with a task. */
 object Task {
 
+  trait TaskInfoLike extends dagr.api.models.TaskInfoLike with Ordered[TaskInfoLike] {
+    /** Gets the associated task. */
+    def task: Task
+
+    def compare(that: TaskInfoLike): Int = {
+      (this.id, that.id) match {
+        case (Some(_), None)    => -1
+        case (None, Some(_))    => 1
+        case (None, None)       => this.status.ordinal - that.status.ordinal
+        case (Some(l), Some(r)) => (l - r).toInt
+      }
+    }
+
+    def scriptPath: Option[FilePath] = this.script.map(PathUtil.pathTo(_))
+    def logPath: Option[FilePath]    = this.script.map(PathUtil.pathTo(_))
```

Comment:
Only needed because `script` is a type `Option[String]`.  I could make a `lazy va`l.